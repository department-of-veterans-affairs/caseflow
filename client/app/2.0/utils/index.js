/* eslint-disable  */
export * from 'utils/reader/index';

/* eslint-enable  */
