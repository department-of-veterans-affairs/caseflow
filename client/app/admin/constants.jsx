export const ACTIONS = {
  SET_FEATURE_TOGGLES: 'SET_FEATURE_TOGGLES',
  POST_VETERAN_EXTRACT: 'POST_VETERAN_EXTRACT',
  STARTED_VETERAN_EXTRACT: 'STARTED_VETERAN_EXTRACT',
  FAILED_VETERAN_EXTRACT: 'FAILED_VETERAN_EXTRACT'
};
