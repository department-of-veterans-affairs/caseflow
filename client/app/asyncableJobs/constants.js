export const ACTIONS = {
  MANUAL_JOB_STARTED: 'MANUAL_JOB_STARTED',
  MANUAL_JOB_FAILED: 'MANUAL_JOB_FAILED',
  CLEAR_MANUAL_JOB_TYPE: 'CLEAR_MANUAL_JOB_TYPE',
};