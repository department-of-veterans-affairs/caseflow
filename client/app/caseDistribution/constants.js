import DISTRIBUTION from '../../constants/DISTRIBUTION';

const LEVERS = 'levers';

export const Constant = {
  LEVERS
};

export const sectionTitles = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]: DISTRIBUTION.ama_hearings_section_title,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]: DISTRIBUTION.ama_direct_review_section_title,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]: DISTRIBUTION.ama_evidence_submission_section_title
};

export const docketTimeGoalPriorMappings = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_hearings_docket_time_goals,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_direct_review_docket_time_goals,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_evidence_submission_docket_time_goals
};
