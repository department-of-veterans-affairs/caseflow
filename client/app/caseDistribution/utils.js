import ACD_LEVERS from '../../constants/ACD_LEVERS';

export const findOption = (lever, value) => lever.options.find((option) => option.item === value);

export const createCombinationValue = (toggleValue, value) => `${toggleValue}-${value}`;

/**
 * Add backendValue attributes to each lever
 * For radio and combination levers add currentValue
 */
export const createUpdatedLeversWithValues = (levers) => {
  const leverGroups = Object.keys(levers);

  const leversWithValues = () => {
    return leverGroups.reduce((updatedLevers, leverGroup) => {
      updatedLevers[leverGroup] = levers[leverGroup].map((lever) => {
        let additionalValues = {
          backendValue: lever.value
        };

        const dataType = lever.data_type;

        // Only add a new property for radio and combination data types as these have special handling logic
        // to retrieve value
        if (dataType === ACD_LEVERS.data_types.radio) {
          additionalValues = {
            currentValue: findOption(lever, lever.value).value,
            backendValue: findOption(lever, lever.value).value,
          };
        } else if (dataType === ACD_LEVERS.data_types.combination) {
          additionalValues = {
            currentValue: createCombinationValue(lever.is_toggle_active, lever.value),
            backendValue: createCombinationValue(lever.is_toggle_active, lever.value)
          };
        }

        return {
          ...lever,
          ...additionalValues
        };
      });

      return updatedLevers;
    }, {});
  };

  return leversWithValues();
};

export const formatTimestamp = (entry) => {
  const dateEntry = new Date(entry);
  const timeEntry = {
    hours: `0${dateEntry.getHours()}`.slice(-2),
    minutes: `0${dateEntry.getMinutes()}`.slice(-2),
    seconds: `0${dateEntry.getSeconds()}`.slice(-2),
  };

  const formattedDate = `${dateEntry.toLocaleDateString('en-US', {
    weekday: 'short',
    year: 'numeric',
    month: 'short',
    day: 'numeric'

  }) } ${timeEntry.hours}:${timeEntry.minutes}:${timeEntry.seconds}`;

  return formattedDate;
};

export const formatLeverHistory = (leverHistoryList) => {

  if (!leverHistoryList) {
    return [];
  }

  const formattedHistory = leverHistoryList.reduce((accumulator, entry) => {
    const existingEntry = accumulator.find(
      (item) => formatTimestamp(item.created_at) === formatTimestamp(entry.created_at) &&
      item.user_css_id === entry.user_css_id
    );

    if (existingEntry) {
      existingEntry.titles.push(entry.lever_title);
      existingEntry.previous_values.push(entry.previous_value);
      existingEntry.updated_values.push(entry.update_value);
      existingEntry.units.push(entry.lever_unit || 'null');
    } else {
      const newEntry = {
        created_at: formatTimestamp(entry.created_at),
        user_css_id: entry.user_css_id,
        user_name: entry.user_name,
        titles: [entry.lever_title],
        previous_values: [entry.previous_value],
        updated_values: [entry.update_value],
        units: [entry.lever_unit || 'null'],
      };

      accumulator.push(newEntry);
    }

    return accumulator;
  }, []);

  let descendingFormattedHistory = formattedHistory.reverse();

  return descendingFormattedHistory;
};

export const validateLeverInput = (lever, value) => {
  const errors = [];
  const { min_value: minValue, max_value: maxValue } = lever;

  if (value === null || value === '') {
    errors.push({ leverItem: lever.item, message: ACD_LEVERS.validation_error_message.minimum_not_met });
  }
  if (parseFloat(value)) {
    if (value < minValue) {
      errors.push({ leverItem: lever.item, message: ACD_LEVERS.validation_error_message.minimum_not_met });
    }
    if (maxValue && value > maxValue) {
      errors.push({ leverItem: lever.item, message: ACD_LEVERS.validation_error_message.out_of_bounds });
    }
  }

  return errors;
};

export const leverErrorMessageExists = (existingErrors, newErrors) => {
  return existingErrors.some((existingError) =>
    newErrors.every((newError) =>
      JSON.stringify(existingError) === JSON.stringify(newError)
    )
  );
};

export const dynamicallyAddAsterisk = (lever) => {
  return (lever.algorithms_used.includes(ACD_LEVERS.algorithms.proportion) &&
    lever.algorithms_used.includes(ACD_LEVERS.algorithms.docket) ? '*' : '');
};
