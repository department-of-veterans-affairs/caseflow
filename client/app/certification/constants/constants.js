// actions
//
export const UPDATE_PROGRESS_BAR = 'UPDATE_PROGRESS_BAR';
export const TOGGLE_CANCELLATION_MODAL = 'TOGGLE_CANCELLATION_MODAL';
export const CHANGE_VBMS_HEARING_DOCUMENT = 'CHANGE_VBMS_HEARING_DOCUMENT';
export const CHANGE_TYPE_OF_FORM9 = 'CHANGE_TYPE_OF_FORM9';
export const CHANGE_TYPE_OF_HEARING = 'CHANGE_TYPE_OF_HEARING';

export const CHANGE_REPRESENTATIVE_NAME = 'CHANGE_REPRESENTATIVE_NAME';
export const CHANGE_REPRESENTATIVE_TYPE = 'CHANGE_REPRESENTATIVE_TYPE';
export const CHANGE_OTHER_REPRESENTATIVE_TYPE = 'CHANGE_OTHER_REPRESENTATIVE_TYPE';

export const CHANGE_SIGN_AND_CERTIFY_FORM = 'CHANGE_SIGN_AND_CERTIFY_FORM';

export const CERTIFICATION_UPDATE_REQUEST = 'CERTIFICATION_UPDATE_REQUEST';
export const CERTIFICATION_UPDATE_FAILURE = 'CERTIFICATION_UPDATE_FAILURE';
export const CERTIFICATION_UPDATE_SUCCESS = 'CERTIFICATION_UPDATE_SUCCESS';

export const ON_CONTINUE_CLICK_FAILED = 'ON_CONTINUE_CLICK_FAILED';
export const ON_CONTINUE_CLICK_SUCCESS = 'ON_CONTINUE_CLICK_SUCCESS';

export const RESET_STATE = 'RESET_STATE';

// types of hearings
//
// TODO:
// on the backend, NO_HEARING_DESIRED
// should result in VIDEO being written to VACOLS,
// and HEARING_CANCELLED should result in a cancellation
// checkbox being checked, but the original hearing type
// should be undisturbed.
export const hearingPreferences = {
  VIDEO: 'VIDEO',
  TRAVEL_BOARD: 'TRAVEL_BOARD',
  WASHINGTON_DC: 'WASHINGTON_DC',
  HEARING_TYPE_NOT_SPECIFIED: 'HEARING_TYPE_NOT_SPECIFIED',
  NO_HEARING_DESIRED: 'NO_HEARING_DESIRED',
  HEARING_CANCELLED: 'HEARING_CANCELLED',
  NO_BOX_SELECTED: 'NO_BOX_SELECTED'
};

// form9 values
export const form9Types = {
  FORMAL_FORM9: 'FORMAL_FORM9',
  INFORMAL_FORM9: 'INFORMAL_FORM9'
};

// representation for the appellant
export const representativeTypes = {
  ATTORNEY: 'ATTORNEY',
  AGENT: 'AGENT',
  ORGANIZATION: 'ORGANIZATION',
  NONE: 'NONE',
  // TODO: should "Other be a real type"?
  OTHER: 'OTHER'
};

// was a hearing document found in VBMS?
export const vbmsHearingDocument = {
  FOUND: 'FOUND',
  NOT_FOUND: 'NOT_FOUND'
};

export const progressBarSections = {
  CHECK_DOCUMENTS: 'CHECK_DOCUMENTS',
  CONFIRM_CASE_DETAILS: 'CONFIRM_CASE_DETAILS',
  CONFIRM_HEARING: 'CONFIRM_HEARING',
  SIGN_AND_CERTIFY: 'SIGN_AND_CERTIFY'
};

export const certifyingOfficialTitles = {
  DECISION_REVIEW_OFFICER: 'DECISION_REVIEW_OFFICER',
  RATING_SPECIALIST: 'RATING_SPECIALIST',
  VETERANS_SERVICE_REPRESENTATIVE: 'VETERANS_SERVICE_REPRESENTATIVE',
  CLAIMS_ASSISTANT: 'CLAIMS_ASSISTANT',
  OTHER: 'OTHER'
};
