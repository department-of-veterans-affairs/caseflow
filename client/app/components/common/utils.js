export const consolidatedEmptyValuesFor = (val) => {
  return [undefined, null, 'undefined', 'null'].includes(val) ? 'null' : val;
}
