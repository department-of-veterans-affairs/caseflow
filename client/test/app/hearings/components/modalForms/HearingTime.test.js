import React from 'react';

import { HearingTime } from 'app/hearings/components/modalForms/HearingTime';
import { ReadOnly } from 'app/hearings/components/details/ReadOnly';
import { mount } from 'enzyme';
import moment from 'moment-timezone/moment-timezone';
import HEARING_TIME_OPTIONS from 'constants/HEARING_TIME_OPTIONS';
import { COMMON_TIMEZONES } from 'app/constants/AppConstants';
import TIMEZONES from 'constants/TIMEZONES';
import { shortZoneName } from 'app/hearings/utils';

const [timezoneLabel] = Object.keys(TIMEZONES).filter((zone) => TIMEZONES[zone] === COMMON_TIMEZONES[3]);

describe('HearingTime', () => {
  // Ignore warnings about SearchableDropdown
  jest.spyOn(console, 'error').mockReturnValue();

  test('Matches snapshot with default props when passed in', () => {
    const form = mount(<HearingTime />);

    expect(form).toMatchSnapshot();

    const checkedRadio = form.find('input').find({ checked: true });

    // A single input is checked by default, and it's the "Other" radio
    expect(checkedRadio.exists()).toBe(true);
    expect(checkedRadio.exists({ value: 'Other' })).toBe(true);

    const dropdown = form.find('Select');

    // The select field is not disabled by default (meaning the "Other")
    // radio is checked
    expect(dropdown.exists()).toBe(true);
    expect(dropdown.exists({ disabled: false })).toBe(true);

    // Expect the naming of forms to match expected
    expect(form.exists({ name: 'hearingTime0' })).toBe(true);
    expect(form.exists({ name: 'optionalHearingTime0' })).toBe(true);
  });

  test('Matches snapshot when enableZone is true', () => {
    // Run the test
    const hearingTime = mount(<HearingTime enableZone value={HEARING_TIME_OPTIONS[0].value} />);

    // Assertions
    expect(hearingTime).toMatchSnapshot();
    expect(hearingTime.find('Select').prop('value').label).toContain(timezoneLabel);
  });

  test('Matches snapshot when other time is not selected', () => {
    const form = mount(<HearingTime enableZone value="12:30 PM Eastern Time (US & Canada)" />);

    expect(form).toMatchSnapshot();

    expect(form.find('input').exists({ checked: true, value: '12:30 PM Eastern Time (US & Canada)' })).toBe(
      true
    );
  });

  test('Matches snapshot when other time is selected', () => {
    const selectedTime = '1:45 PM Eastern Time (US & Canada)';
    const option = HEARING_TIME_OPTIONS.find(({ value }) => value === selectedTime);

    const form = mount(<HearingTime value={selectedTime} />);

    expect(form).toMatchSnapshot();

    // Expect "Other" radio to be checked
    expect(form.find('input').exists({ checked: true, value: 'Other' })).toBe(true);

    // Expect dropdown to be populated with correct time
    expect(form.exists('SearchableDropdown')).toBe(true);
    expect(form.find('Select').exists({ value: option })).toBe(true);
  });

  test('Matches snapshot when readonly prop is set', () => {
    const form = mount(<HearingTime />);

    expect(form).toMatchSnapshot();

    expect(form.find('select').every({ disabled: true })).toBe(true);
  });
});
