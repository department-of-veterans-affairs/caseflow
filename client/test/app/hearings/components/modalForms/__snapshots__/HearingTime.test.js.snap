// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HearingTime Matches snapshot when enableZone is true 1`] = `
<DocumentFragment>
  <span
    data-css-1042gbc=""
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_Invalid dateEastern Time (US & Canada)"
            name="hearingTime0"
            type="radio"
            value="Invalid dateEastern Time (US & Canada)"
          />
          <label
            class=""
            for="hearingTime0_Invalid dateEastern Time (US & Canada)"
          >
            Invalid dateEastern Time (US & Canada)
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_Invalid dateEastern Time (US & Canada)"
            name="hearingTime0"
            type="radio"
            value="Invalid dateEastern Time (US & Canada)"
          />
          <label
            class=""
            for="hearingTime0_Invalid dateEastern Time (US & Canada)"
          >
            Invalid dateEastern Time (US & Canada)
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            id="hearingTime0_Other"
            name="hearingTime0"
            type="radio"
            value="Other"
          />
          <label
            class=""
            for="hearingTime0_Other"
          >
            Other
          </label>
        </div>
      </div>
    </fieldset>
  </span>
  <div
    class=""
  >
    <div
      class="cf-form-dropdown dropdown-optionalHearingTime0"
    >
      <label
        class="question-label usa-sr-only"
        for="optional-hearing-time-0"
        id="optional-hearing-time-0-label"
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </label>
      <div
        class="cf-select"
      >
        <div
          class=" css-2b097c-container"
        >
          <div
            class="cf-select__control css-yk16xz-control"
          >
            <div
              class="cf-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="cf-select__placeholder css-1wa3eu0-placeholder"
              >
                Select a time
              </div>
              <div
                class="css-81nyic-Input"
              >
                <div
                  class="cf-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-labelledby="optional-hearing-time-0-label"
                    aria-owns="optional-hearing-time-0-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="optional-hearing-time-0"
                    role="combobox"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="optionalHearingTime0"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HearingTime Matches snapshot when other time is not selected 1`] = `
<DocumentFragment>
  <span
    data-css-1042gbc=""
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_8:30 AM"
            name="hearingTime0"
            type="radio"
            value="8:30 AM"
          />
          <label
            class=""
            for="hearingTime0_8:30 AM"
          >
            8:30 AM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_12:30 PM"
            name="hearingTime0"
            type="radio"
            value="12:30 PM"
          />
          <label
            class=""
            for="hearingTime0_12:30 PM"
          >
            12:30 PM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            id="hearingTime0_Other"
            name="hearingTime0"
            type="radio"
            value="Other"
          />
          <label
            class=""
            for="hearingTime0_Other"
          >
            Other
          </label>
        </div>
      </div>
    </fieldset>
  </span>
  <div
    class=""
  >
    <div
      class="cf-form-dropdown dropdown-optionalHearingTime0"
    >
      <label
        class="question-label usa-sr-only"
        for="optional-hearing-time-0"
        id="optional-hearing-time-0-label"
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </label>
      <div
        class="cf-select"
      >
        <div
          class=" css-2b097c-container"
        >
          <div
            class="cf-select__control css-yk16xz-control"
          >
            <div
              class="cf-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="cf-select__placeholder css-1wa3eu0-placeholder"
              >
                Select a time
              </div>
              <div
                class="css-81nyic-Input"
              >
                <div
                  class="cf-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-labelledby="optional-hearing-time-0-label"
                    aria-owns="optional-hearing-time-0-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="optional-hearing-time-0"
                    role="combobox"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="optionalHearingTime0"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HearingTime Matches snapshot when other time is selected 1`] = `
<DocumentFragment>
  <span
    data-css-1042gbc=""
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_8:30 AM"
            name="hearingTime0"
            type="radio"
            value="8:30 AM"
          />
          <label
            class=""
            for="hearingTime0_8:30 AM"
          >
            8:30 AM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_12:30 PM"
            name="hearingTime0"
            type="radio"
            value="12:30 PM"
          />
          <label
            class=""
            for="hearingTime0_12:30 PM"
          >
            12:30 PM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            id="hearingTime0_Other"
            name="hearingTime0"
            type="radio"
            value="Other"
          />
          <label
            class=""
            for="hearingTime0_Other"
          >
            Other
          </label>
        </div>
      </div>
    </fieldset>
  </span>
  <div
    class=""
  >
    <div
      class="cf-form-dropdown dropdown-optionalHearingTime0"
    >
      <label
        class="question-label usa-sr-only"
        for="optional-hearing-time-0"
        id="optional-hearing-time-0-label"
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </label>
      <div
        class="cf-select"
      >
        <div
          class=" css-2b097c-container"
        >
          <div
            class="cf-select__control css-yk16xz-control"
          >
            <div
              class="cf-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="cf-select__placeholder css-1wa3eu0-placeholder"
              >
                Select a time
              </div>
              <div
                class="css-81nyic-Input"
              >
                <div
                  class="cf-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-labelledby="optional-hearing-time-0-label"
                    aria-owns="optional-hearing-time-0-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="optional-hearing-time-0"
                    role="combobox"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="optionalHearingTime0"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HearingTime Matches snapshot when readonly prop is set 1`] = `
<DocumentFragment>
  <span
    data-css-1042gbc=""
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            disabled=""
            id="hearingTime0_8:30 AM"
            name="hearingTime0"
            type="radio"
            value="8:30 AM"
          />
          <label
            class="disabled"
            for="hearingTime0_8:30 AM"
          >
            8:30 AM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            disabled=""
            id="hearingTime0_12:30 PM"
            name="hearingTime0"
            type="radio"
            value="12:30 PM"
          />
          <label
            class="disabled"
            for="hearingTime0_12:30 PM"
          >
            12:30 PM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            disabled=""
            id="hearingTime0_Other"
            name="hearingTime0"
            type="radio"
            value="Other"
          />
          <label
            class="disabled"
            for="hearingTime0_Other"
          >
            Other
          </label>
        </div>
      </div>
    </fieldset>
  </span>
  <div
    class=""
  >
    <div
      class="cf-form-dropdown dropdown-optionalHearingTime0"
    >
      <label
        class="question-label usa-sr-only"
        for="optional-hearing-time-0"
        id="optional-hearing-time-0-label"
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </label>
      <div
        class="cf-select"
      >
        <div
          class="cf-select--is-disabled css-14jk2my-container"
        >
          <div
            class="cf-select__control cf-select__control--is-disabled css-1fhf3k1-control"
          >
            <div
              class="cf-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="cf-select__placeholder css-1wa3eu0-placeholder"
              >
                Select a time
              </div>
              <input
                aria-autocomplete="list"
                class="css-62g3xt-dummyInput"
                disabled=""
                id="optional-hearing-time-0"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="cf-select__indicator-separator css-109onse-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`HearingTime Matches snapshot with default props when passed in 1`] = `
<DocumentFragment>
  <span
    data-css-1042gbc=""
  >
    <fieldset
      class="usa-fieldset-inputs cf-form-radio "
    >
      <legend
        class=""
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </legend>
      <div
        class="cf-form-radio-options"
      >
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_8:30 AM"
            name="hearingTime0"
            type="radio"
            value="8:30 AM"
          />
          <label
            class=""
            for="hearingTime0_8:30 AM"
          >
            8:30 AM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            id="hearingTime0_12:30 PM"
            name="hearingTime0"
            type="radio"
            value="12:30 PM"
          />
          <label
            class=""
            for="hearingTime0_12:30 PM"
          >
            12:30 PM
          </label>
        </div>
        <div
          class="cf-form-radio-option"
        >
          <input
            checked=""
            id="hearingTime0_Other"
            name="hearingTime0"
            type="radio"
            value="Other"
          />
          <label
            class=""
            for="hearingTime0_Other"
          >
            Other
          </label>
        </div>
      </div>
    </fieldset>
  </span>
  <div
    class=""
  >
    <div
      class="cf-form-dropdown dropdown-optionalHearingTime0"
    >
      <label
        class="question-label usa-sr-only"
        for="optional-hearing-time-0"
        id="optional-hearing-time-0-label"
      >
        <strong>
          <span>
            Hearing Time
          </span>
        </strong>
      </label>
      <div
        class="cf-select"
      >
        <div
          class=" css-2b097c-container"
        >
          <div
            class="cf-select__control css-yk16xz-control"
          >
            <div
              class="cf-select__value-container css-g1d714-ValueContainer"
            >
              <div
                class="cf-select__placeholder css-1wa3eu0-placeholder"
              >
                Select a time
              </div>
              <div
                class="css-81nyic-Input"
              >
                <div
                  class="cf-select__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-labelledby="optional-hearing-time-0-label"
                    aria-owns="optional-hearing-time-0-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="optional-hearing-time-0"
                    role="combobox"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="optionalHearingTime0"
            type="hidden"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
